### ISSUE TITLE
hasMany: destroyAssociations

### ISSUE THREAD

**Issue by kristijanhusak** [sequelize/sequelize#6095]
Hello,

I'm trying to figure out if there is a way to remove associations from table instead of setting foreign key to null when i call `setAssociations(null)`.
For example, if i have Author and Post models:

``` js
const Author = Sequelize.define('author', {
  name: Sequelize.STRING
});

const Post = Sequelize.define('post', {
  title: Sequelize.STRING
});

Author.hasMany(Post);
Post.belongsTo(Author);
```

And i want to remove all posts for this author, I would like to use something like this:

``` js
const author = Author.findById(1);
author.setPosts([]).then(() => ...);
```

But the problem is that this just sets `authorId` to NULL in `posts` table. How can i literally remove it from the table?

Thanks.

**Comment by mickhansen** [comment]
Not currently possible. Might make sense to add a `destroyPosts` method

**Comment by kristijanhusak** [comment]
Yeah that would be useful, because setting null on HasMany/BelongsTo relationship does not make sense, only if `paranoid` option is used.

**Comment by mickhansen** [comment]
@kristijanhusak In some cases it does. Personally i've never had a case where i wanted to remove the entire set of associations if i wasnt removing the parent aswell.

**Comment by kristijanhusak** [comment]
I don't need to remove entire set, just put it here as an example.
I want to be able to delete "unused" ones when i'm updating the Author.
So if he initially had posts with id 1, 2, and 3, and if i pass `setPosts([1, 2])`, I expect post with id of 3 to be deleted, not unassociated. It makes sense with ManyToMany, but OneToMany not really.

**Comment by mickhansen** [comment]
@kristijanhusak In your case it doesn't make sense - But there are other cases where it does.

**Comment by kristijanhusak** [comment]
Of course :) Thanks for the answer. 
Anyway, `destroyAssociation` would be useful to have.

**Comment by mickhansen** [comment]
Agreed - If you're looking for it in the short term you'll likely have to submit a PR - We're quite busy with 4.0 refactoring at the moment.

**Comment by kristijanhusak** [comment]
I've setup a workaround for now, and unfortunately does not have time to work on a PR, so i'll wait for now.

**Comment by felixfbecker** [comment]
@kristijanhusak What happens when you pass `{foreignKey: {allowNull: false}}` as second argument to `belongsTo()` and `hasMany()`?

**Comment by afucher** [comment]
Hi, I was searching for that... Already asked a [question](http://stackoverflow.com/questions/41143148/how-to-delete-all-register-from-an-mn-association-table-in-sequelize) in stackoverflow about this...
I want to try to send a PR, could someone tell me wich branch should I take as a base to make this changes? Master? @mickhansen could you help me with that?

**Comment by felixfbecker** [comment]
master :)

**Comment by kesemdavid** [comment]
@kristijanhusak @sushantdhiman Is this issue already resolved and in master by now? looking for the exact same thing

**Comment by chrys1110** [comment]
Can anyone give me example how to workaround this issue with **destroy**?

**Comment by kesemdavid** [comment]
@chrys1110 
My workaround was 
`Model.setAssociations([])` before updating, causing all the associations to be destroyed using the instance you have.

For example : `User.setProjects([])`

**Comment by shivasurya** [comment]
any updates ? please
